bsnes/target-libretro/libretro.cpp

```diff
@@ -624,10 +644,14 @@ static void run_with_runahead(const int frames)
 	emulator->unserialize(state);
 }
 
-RETRO_API void retro_run()
+void retro_run()
 {
-	check_variables();
+	bool updated = false;
+	if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE, &updated) && updated)
+		update_variables();
+
 	input_poll();
+	update_geometry();
```

Comment:
this api should be called as minimal as possible, like if there is indeed an actual change in width/height/aspect etc.

Reply:
Yes, I see what you mean. If I understand the code correctly, previously what was happening was:

- retro_run called check_variables (every frame?);
- check_variables called flush_variables only if the core options values were updated;
- flush_variables included the code at the end that would refresh the geometry.

Instead, now that I decoupled update_geometry from update_variables and put it directly under retro_run, it is being called always regardless of whether the core options values have changed. 

So would it suffice to put update_geometry under the “updated” conditional in retro_run? That would, to the best of my knowledge, restore the original behavior (geometry updated only when core options are updated).

Reply:
I have just pushed a commit that should accommodate your request: now update_geometry should be called exclusively when core options are updated. Let me know if this is a suitable approach!